script.skin.helper.service/plugin.py

```diff
@@ -97,27 +125,28 @@ def __init__(self):
 
         else:
             #do plugin main listing...
-            plugincontent.doMainListing(params.get("content_type",[""])[0])
+            plugincontent.doMainListing()
 
 if (__name__ == "__main__"):
     try:
-        if not utils.WINDOW.getProperty("SkinHelperShutdownRequested"):
-            if enableProfiling:
-                import cProfile
-                import pstats
-                import random
-                from time import gmtime, strftime
-                filename = os.path.join( ADDON_DATA_PATH, strftime( "%Y%m%d%H%M%S",gmtime() ) + "-" + str( random.randrange(0,100000) ) + ".log" )
-                cProfile.run( 'Main()', filename )
-                stream = open( filename + ".txt", 'w')
-                stream.write(sys.argv[2])
-                p = pstats.Stats( filename, stream = stream )
-                p.sort_stats( "cumulative" )
-                p.print_stats()
-            else:
-                Main()
-        else:
+        if utils.WINDOW.getProperty("SkinHelperShutdownRequested"):
             utils.logMsg("plugin.py --> Not forfilling request: Kodi is exiting" ,0)
+            xbmcplugin.endOfDirectory(handle=int(sys.argv[1]))
+        elif enableProfiling:
+            import cProfile
+            import pstats
+            import random
+            from time import gmtime, strftime
+            filename = os.path.join( ADDON_DATA_PATH, strftime( "%Y%m%d%H%M%S",gmtime() ) + "-" + str( random.randrange(0,100000) ) + ".log" )
+            cProfile.run( 'Main()', filename )
+            stream = open( filename + ".txt", 'w')
```

Comment:
Good catch, in fact this is just my debug code I use from time to time to check my code for speed. It will never be used by users. Want me to change it right away ?